### ISSUE TITLE
[pod-gateway] 

### ISSUE THREAD

**Issue by Emiller88** [k8s-at-home/charts#1546]
### Helm chart name

pod-gateway

### Helm chart version

5.3.0

### Container name

ghcr.io/k8s-at-home/pod-gateway

### Container tag

v1.5.0

### Description

Introduced by https://github.com/k8s-at-home/charts/pull/1540

For some reason, it broke my deployment. I haven't gotten to the bottom of this one. I think it has to do with the UDP addition.

### Expected result

The previous config should still function.

### Helm values to reproduce

```YAML
image:
      repository: ghcr.io/k8s-at-home/pod-gateway
      tag: v1.5.0

    routed_namespaces:
      - downloads

    publicPorts:
      - hostname: qbittorrent # hostname assigned to the pod
        IP: 10 # must be an integer between 2 and VXLAN_GATEWAY_FIRST_DYNAMIC_IP (20 by default)
        ports:
          - type: udp
            port: ${VPN_FORWARDED_PORT_1}
          - type: tcp
            port: ${VPN_FORWARDED_PORT_1}

    settings:
      NOT_ROUTED_TO_GATEWAY_CIDRS: '${NETWORK_K8S_CLUSTER_CIDR} ${NETWORK_K8S_SERVICE_CIDR} ${NETWORK_MANAGEMENT_CIDR}'
      VPN_INTERFACE: wg0
      VPN_BLOCK_OTHER_TRAFFIC: true
      VPN_TRAFFIC_PORT: ${VPN_GATEWAY_VPN_PORT}
      VPN_LOCAL_CIDRS: '${NETWORK_K8S_CLUSTER_CIDR} ${NETWORK_K8S_SERVICE_CIDR} ${NETWORK_MANAGEMENT_CIDR}'

    webhook:
      image:
        repository: ghcr.io/k8s-at-home/gateway-admision-controller
        tag: v3.4.0

      gatewayDefault: false
      gatewayLabel: setGateway
      gatewayAnnotation: setGateway

    livenessProbe:
      exec:
        command:
          - sh
          - -c
          - if [ $(wget -q -O- https://ipinfo.io/country) == 'US' ]; then exit 0; else exit $?; fi
      initialDelaySeconds: 30
      periodSeconds: 60
      failureThreshold: 1

    addons:
      vpn:
        enabled: true
        type: wireguard

        wireguard:
          image:
            repository: ghcr.io/k8s-at-home/wireguard
            tag: v1.0.20210914
            pullPolicy: IfNotPresent

        securityContext:
          runAsUser: 568
          runAsGroup: 568

        env:
          TZ: '${TIMEZONE}'
          IPTABLES_BACKEND: nft
          KILLSWITCH: 'false'

        configFileSecret: vpn-gateway-vpnconfig

        resources:
          requests:
            cpu: '0.001'
            memory: 128Mi
          limits:
            memory: 512Mi

        networkPolicy:
          enabled: true

          egress:
            - to:
                - ipBlock:
                    cidr: 0.0.0.0/0
              ports:
                - port: ${VPN_GATEWAY_VPN_PORT}
                  protocol: UDP
            - to:
                - ipBlock:
                    cidr: ${NETWORK_K8S_CLUSTER_CIDR}
                - ipBlock:
                    cidr: ${NETWORK_K8S_SERVICE_CIDR}
```

### Repo link

https://github.com/Emiller88/home-cluster/blob/2e9fd5b931aa895a514eee1d06f6219de08dafa1/cluster/apps/vpn/vpn-gateway/helm-release.yaml

**Comment by angelnu** [comment]
@Emiller88 - if you would like to use port 443 with UDP (the old default) you would also need to set the variable `VPN_TRAFFIC_PORT`:

```
    settings:
      NOT_ROUTED_TO_GATEWAY_CIDRS: '${NETWORK_K8S_CLUSTER_CIDR} ${NETWORK_K8S_SERVICE_CIDR} ${NETWORK_MANAGEMENT_CIDR}'
      VPN_INTERFACE: wg0
      VPN_BLOCK_OTHER_TRAFFIC: true
      VPN_TRAFFIC_PORT: ${VPN_GATEWAY_VPN_PORT}
      VPN_LOCAL_CIDRS: '${NETWORK_K8S_CLUSTER_CIDR} ${NETWORK_K8S_SERVICE_CIDR} ${NETWORK_MANAGEMENT_CIDR}'
      VPN_TRAFFIC_PORT: 443
```

**Comment by Emiller88** [comment]
But that shouldn't affect it, I've already set `VPN_TRAFFIC_PORT`
```
      VPN_TRAFFIC_PORT: ${VPN_GATEWAY_VPN_PORT}
```
It's also set to the same port in the egress. I think there was a bug prior maybe, because it doesn't look like any of the logic to override the `VPN_TRAFFIC_PORT` changed.

**Comment by angelnu** [comment]
oh, I did not see that x-)

Did it break only when you updated the Helm chart or did you change anything else in parallel? I released some updates for the associated docker containers that might be more problematic.

The Helm chart only changed the default values that you anyway override so I cannot think of a reason why it would be the cause.

**Comment by Emiller88** [comment]
That's a good question, let me try reverting the container first, then revert the helm chart. I missed that the container got updated as well so that could be it(https://github.com/k8s-at-home/pod-gateway/commit/bc1bd448e84e427b12e87ed8f68191188e370e0b)

**Comment by Emiller88** [comment]
Yep it was the image update! Feel free to close and make a new issue/move this one to that repo!

**Comment by angelnu** [comment]
Good - hopefully you can find why the new container breaks your setup - the main change was https://github.com/k8s-at-home/pod-gateway/pull/22
